dexmaker/src/main/java/com/google/dexmaker/stock/ProxyBuilder.java

```diff
@@ -630,6 +631,43 @@ private static void check(boolean condition, String message) {
         for (MethodSetEntry entry : methodsToProxy) {
             results[i++] = entry.originalMethod;
         }
+
+        // Sort the results array so that they are returned by this method
+        // in a deterministic fashion.
+        Arrays.sort(results, new Comparator<Method>() {
+            @Override
+            public int compare(Method method1, Method method2) {
+                MethodId<? extends T, ?> methodId1 = getMethodId(method1);
+                MethodId<? extends T, ?> methodId2 = getMethodId(method2);
+                String methodStr1 = methodId1.toString();
+                String methodStr2 = methodId2.toString();
+                if (methodStr1.equals(methodStr2)) {
```

Comment:
Since Method.toString() captures all the information contained in a method signature and thereby is guaranteed to generate unique results for distinct methods, I think it makes sense to simplify the comparator to return method1.toString().compareTo(method2.toString()).